__tests__/components/badges/RewardBadges.test.tsx

```diff
@@ -0,0 +1,30 @@
+import "@testing-library/jest-dom";
+import RewardBadge from "@/components/badges/RewardBadge";
+import { render, screen } from "@testing-library/react";
+import Coin from "@/components/ui/Coin";
+
+describe("RewardBadges", () => {
+  it("should render  rewardBadges", () => {
+    render(<RewardBadge />);
+    const rewardBadge = screen.getByTestId("RewardBadge");
+    expect(rewardBadge).toBeInTheDocument();
+  });
+
+  it("should render rewardBadges with coins", () => {
+    render(<RewardBadge />);
+    render(<Coin />);
```

Comment:
Avoid rendering the `Coin` component separately as it is already rendered within the `RewardBadge` component 
Something like:
```suggestion
      it("should render rewardBadges with Coin when the token is provided", () => {
    render(<RewardBadge reward={{ token: "BTC" }} />);
    const rewardBadge = screen.queryByTestId("RewardBadge");
    const coin = screen.getByTestId("coin");
    expect(rewardBadge).toBeInTheDocument();
    expect(coin).toBeInTheDocument();
  });
```